project/src/main/java/com/google/sps/data/ProductSearchLibrary.java

```diff
@@ -250,4 +199,94 @@ public static void removeProductFromProductSet(
     return products;
   }
 
-}
\ No newline at end of file
+  public static ArrayList<ProductItem> listProductsInProductSet(
+    String projectId, 
+    String computeRegion, 
+    String productSetId) throws IOException {
+    ArrayList<ProductItem> productItems = new ArrayList<>();
+    try (ProductSearchClient client = ProductSearchClient.create()) {
+
+        // Get the full path of the product set.
+        String formattedName =
+            ProductSearchClient.formatProductSetName(projectId, computeRegion, productSetId);
+        // List all the products available in the product set.
+        for (Product product : client.listProductsInProductSet(formattedName).iterateAll()) {
+            ProductItem productItem = new ProductItem(product.getName().substring(product.getName().lastIndexOf('/') + 1), product.getDisplayName(),
+                                        product.getProductCategory());
+            productItems.add(productItem);
+            for (Product.KeyValue element : product.getProductLabelsList()) {
+                System.out.println(String.format("%s: %s", element.getKey(), element.getValue()));
+            }
+        }
+    } catch(Exception e){
+        System.err.println("Could not list products");
```

Comment:
I would also print out the value of `e` here, in case you need more info for debugging.